cirq-core/cirq/experiments/single_qubit_readout_calibration.py

```diff
@@ -96,19 +98,131 @@ def estimate_single_qubit_readout_errors(
         the probabilities. Also stores a timestamp indicating the time when
         data was finished being collected from the sampler.
     """
+    return estimate_correlated_single_qubit_readout_errors(
+        sampler=sampler,
+        qubits=qubits,
+        repetitions=repetitions,
+        trials=2,
+        bit_strings=np.array([[0, 1] for q in qubits]),
+    )
+
+
+def estimate_correlated_single_qubit_readout_errors(
+    sampler: 'cirq.Sampler',
+    *,
+    qubits: Iterable['cirq.Qid'],
+    trials: int = 20,
+    repetitions: int = 1000,
+    trials_per_batch: Optional[int] = None,
+    bit_strings: np.ndarray = None,
+) -> SingleQubitReadoutCalibrationResult:
+    """Estimate single qubit readout error using parallel operations.
+
+    For each trial, prepare and then measure a random computational basis
+    bitstring on qubits using gates in parallel.
+    Returns a SingleQubitReadoutCalibrationResult which can be used to
+    compute readout errors for each qubit.
+
+    Args:
+        sampler: The `cirq.Sampler` used to run the circuits.
+        qubits: The qubits being tested.
+        repetitions: The number of measurement repetitions to perform for
+            each trial.
+        trials: The number of bitstrings to prepare.
+        trials_per_batch:  If provided, split the experiment into batches
```

Comment:
One other random question: Why is `trials_per_batch` something we want at the experiment level instead of allowing the user to construct a custom `cirq.Sampler` where they can make arbitrary calls to `run_batch` and then behind the scenes it get's broken up into chunks by the sampler ? This feels like a strange place to have something like this am I missing something ?

Reply:
Fair enough, but we don't actually have this sampler right now.  The other thing is that this is a sweep (not a batch), so it's a bit harder to break up.  We could deprecate this parameter once we have a proper batching sampler maybe?